lib/sandthorn/aggregate_root_base.rb

```diff
@@ -83,13 +93,15 @@ def find id
 
         def aggregate_find aggregate_id
           begin
-            events = Sandthorn.find(aggregate_id, self)
-            unless events && !events.empty?
+            aggregate_from_snapshot = Sandthorn.find_snapshot(aggregate_id) if self.snapshot
+            current_aggregate_version = aggregate_from_snapshot.nil? ? 0 : aggregate_from_snapshot.aggregate_current_event_version
+            events = Sandthorn.find(aggregate_id, self, current_aggregate_version)
+            if aggregate_from_snapshot.nil? && events.empty?
```

Comment:
You can write more confident code by wrapping `events` in an array:

```ruby
events = Array(Sandthorn.find(aggregate_id, self, current_aggregate_version)
raise Errors::AggregateNotFound unless events.any?
```

Reply:
I’m sorry, sloppy reading. Nothing to see here!

Reply:
No worries and great thanks for your insights in the language of Ruby.